events/test.py

```diff
@@ -0,0 +1,118 @@
+import unittest
+import datetime
+import app
+import mongomock
+
+
+example_time = datetime.datetime(2019, 6, 11, 10, 33, 1, 100000)
+
+
+class TestEventsDB(unittest.TestCase):
+    def setUp(self):
+        self.event_info = {'name': 'test_event',
+                           'description': 'testing!',
+                           'author': 'admin',
+                           'event_time': example_time,
+                           'created_at': example_time}
+        self.test_event = app.Event(self.event_info)
+        # Create mock DB for testing
+        self.client = mongomock.MongoClient()
+        self.test_coll = self.client.eventsDB.all_events
+
+    def test_add(self):
+        self.test_event.add_to_db(self.test_coll)
+        queried_event = self.test_coll.find_one(self.event_info)
+        self.assertEqual(app.Event(queried_event), self.test_event)
+
+    def test_build_info(self):
+        test_info = {'name': 'test_event',
+                     'description': 'testing!',
+                     'author': 'admin',
+                     'event_time': example_time, }
+        info = app.build_event_info(test_info, example_time)
+        self.assertEqual(info['created_at'], example_time)
+
+    def tearDown(self):
+        self.client.drop_database("eventsDB")
+
+
+class TestEventsClass(unittest.TestCase):
+    def setUp(self):
+        self.test_info = {'name': 'test_event',
+                          'description': 'testing!',
+                          'author': 'admin',
+                          'event_time': example_time,
+                          'created_at': example_time}
+        self.test_info_with_id = dict(event_id=1, **self.test_info)
+        self.test_info_with_db_id = dict(event_id=1, _id=2, **self.test_info)
+
+    def test_construct_event(self):
+        event = app.Event(self.test_info)
+        self.test_info['event_id'] = None
+        self.assertEqual(event.dict, self.test_info)
+
+        event_with_id = app.Event(self.test_info_with_id)
+        self.assertEqual(event_with_id.dict, self.test_info_with_id)
+
+        event_with_db_id = app.Event(self.test_info_with_db_id)
+        self.assertEqual(event_with_db_id.dict, self.test_info_with_db_id)
+
+    def test_constuct_event_error(self):
+        info_missing_name = self.test_info.copy()
+        del info_missing_name['name']
+        info_extra_field = self.test_info.copy()
+        info_extra_field['extra'] = 'hello'
+
+        self.assertRaises(ValueError, app.Event, info_missing_name)
```

Comment:
Prefer the context manager version of assertRaises, as it's easier to read the code (and you can include multiple lines if needed:

```python
with self.assertRaises(ValueError):
  app.Event(info_missing_name)
```

Reply:
Done in https://github.com/knative-portability/large-events/pull/4/commits/d1ade21683ef7f0d5d8c52a230d650691c86cd9f